packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,173 @@
+import '@aws-cdk/assert/jest';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as core from '@aws-cdk/core';
+import * as path from 'path';
+import * as inc from '../lib';
+import * as futils from '../lib/file-utils';
+
+// tslint:disable:object-literal-key-quotes
+/* eslint-disable quotes */
+
+describe('CDK Include', () => {
```

Comment:
For these tests, I feel like we're missing a simple, easy-to-understand base-case tests. All of those go into details of multiple children plus grandchildren. I would love to see something like this:

```ts
  test('correctly modifies the templateUrl property of a nested stack whose template has been passed', () => {
    new inc.CfnInclude(stack, 'ParentStack', {
      templateFile: testTemplateFilePath('only-nested-stack.json'),
      nestedStacks: {
        'NestedStack': {
          // doesn't matter in this test
          templateFile: testTemplateFilePath('child-import-stack.json'),
        },
      },
    });

    const assetBucketParam = 'AssetParameters6b884775090ed88cd1a143f64442a92a6c34eaeff3857976d15ef2e3beee05d7S3BucketC8A1BF52';
    const assetKeyParam = 'AssetParameters6b884775090ed88cd1a143f64442a92a6c34eaeff3857976d15ef2e3beee05d7S3VersionKeyA9E03E19';

    expect(stack).toMatchTemplate({
      "Resources": {
        "NestedStack": {
          "Type": "AWS::CloudFormation::Stack",
          "Properties": {
            "TemplateURL": {
              "Fn::Join": ["", [
                "https://s3.",
                { "Ref": "AWS::Region" },
                ".",
                { "Ref": "AWS::URLSuffix" },
                "/",
                { "Ref": assetBucketParam },
                "/",
                {
                  "Fn::Select": [0,
                    {
                      "Fn::Split": ["||",
                        { "Ref": assetKeyParam },
                      ],
                    },
                  ],
                },
                {
                  "Fn::Select": [1,
                    {
                      "Fn::Split": ["||",
                        { "Ref": assetKeyParam },
                      ],
                    },
                  ],
                },
              ]],
            },
          },
        },
      },
      "Parameters": {
        [assetBucketParam]: {
          "Type": "String",
          "Description": "S3 bucket for asset \"6b884775090ed88cd1a143f64442a92a6c34eaeff3857976d15ef2e3beee05d7\"",
        },
        [assetKeyParam]: {
          "Type": "String",
          "Description": "S3 key for asset version \"6b884775090ed88cd1a143f64442a92a6c34eaeff3857976d15ef2e3beee05d7\"",
        },
        "AssetParameters6b884775090ed88cd1a143f64442a92a6c34eaeff3857976d15ef2e3beee05d7ArtifactHash605B2835": {
          "Type": "String",
          "Description": "Artifact hash for asset \"6b884775090ed88cd1a143f64442a92a6c34eaeff3857976d15ef2e3beee05d7\"",
        },
      },
    });
  });
```

where `only-nested-stack.json` is, as the name suggests:

```json
{
  "Resources": {
    "NestedStack": {
      "Type": "AWS::CloudFormation::Stack",
      "Properties": {
        "TemplateURL": "doesnt-matter"
      }
    }
  }
}
```

And more tests in this simple spirit.